### PR TITLE
refactor: Rename metrics-port to management-port

### LINKED ISSUE #867: Rename "metrics" port to "management" port
Currently, the "metrics" port configuration controls the port the HTTP readyz/healthz checks, gRPC health check, and metrics.

We should change the name to "management" port to make clear it exposes more than just metrics.

Relates to: https://github.com/open-feature/flagd/pull/863

Be sure to update any related doc.

### PR DESCRIPTION
**Pull request by craigpastro**
## This PR

- Renames the flag `--metrics-port` to `--management-port`
- The short flag name `-m` now relates to `--management-port`
- Add a deprecated notice to `--metrics-port`
- Replaces "metrics port" with "management port" in docs (and any variants)
- Behaviour: If `--management-port` is set use that value (and possibly ignore `--metrics-port`). If not and `--metrics-port` is set use that value. Otherwise, use the default value.

### Related Issues

Resolves #867.

### CODE REVIEW

File: docs/reference/openfeature-operator/crds/featureflagsource.md [hunk 1]
```diff
@@ -63,14 +63,14 @@ sources:
...
-| port          | Flag evaluation endpoint port | 8013                       |
-| metricsPort   | Metrics port                  | 8014                       |
-| evaluator     | Evaluator to use              | json                       |
-| image         | flagD image                   | ghcr.io/open-feature/flagd |
-| tag           | flagD image tag               | Latest tag                 |
-| probesEnabled | Enable/Disable health probes  | true                       |
+| Configuration  | Explanation                   | Default                    |
+|----------------|-------------------------------|----------------------------|
+| port           | Flag evaluation endpoint port | 8013                       |
+| managementPort | Management port               | 8014                       |
```
Comment (beeme1mr) [inline docs/reference/openfeature-operator/crds/featureflagsource.md:69]: We'll need to make sure this is updated in OFO. FYI @toddbaert @Kavindu-Dodan @thisthat

**Review by beeme1mr — APPROVED** [review]

File: docs/reference/openfeature-operator/crds/featureflagsource.md (hunk 1 above)
Comment (Kavindu-Dodan) [reply docs/reference/openfeature-operator/crds/featureflagsource.md:69]: Yes, we need a follow-up issue to migrate OFO.

However, current configurations won't break after this fix as @craigpastro did a good job of having a fallback for the deprecated option [^1]. 

[^1]: https://github.com/open-feature/flagd/blob/62033448903b20eb992014e25a365cafe955400e/flagd-proxy/cmd/start.go#L75-L80

**Review by Kavindu-Dodan — APPROVED** [review]
Changes looks good. Like @beeme1mr said, we need to upadte Openfeature Operator to adapt to this configuration change.

If we release flagd, @odubajDT is this something that we could include with beta version changeset ?

**Review by toddbaert — APPROVED** [review]

File: flagd-proxy/cmd/start.go [hunk 2]
```diff
@@ -64,10 +68,21 @@ var startCmd = &cobra.Command{
 
 		syncStore := syncStore.NewSyncStore(ctx, logger)
 		s := syncServer.NewServer(logger, syncStore)
+
+		// If --management-port is set use that value. If not and
+		// --metrics-port is set use that value. Otherwise use the default
+		// value.
+		managementPort := uint16(defaultManagementPort)
+		if viper.GetUint16(managementPortFlagName) != defaultManagementPort {
```
Comment (toddbaert) [inline flagd-proxy/cmd/start.go:76]: Should we add a deprecation log message here? cc @beeme1mr @Kavindu-Dodan

File: flagd-proxy/cmd/start.go [hunk 3]
```diff
@@ -64,10 +68,21 @@ var startCmd = &cobra.Command{
...
+			managementPort = viper.GetUint16(managementPortFlagName)
+		} else if viper.GetUint16(metricsPortFlagName) != defaultManagementPort {
+			managementPort = viper.GetUint16(metricsPortFlagName)
```
Comment (toddbaert) [inline flagd-proxy/cmd/start.go:79]: Is this actually breaking with the change [here](https://github.com/open-feature/flagd/pull/1012/files#diff-b91a595f5d0f13cb46c493699f061cc8242e7cdef20cb73de82a24fe9b7b3b19R72-R79)?

**Review by toddbaert — APPROVED** [review]
Is this actually breaking with the change [here](https://github.com/open-feature/flagd/pull/1012/files#diff-b91a595f5d0f13cb46c493699f061cc8242e7cdef20cb73de82a24fe9b7b3b19R72-R79)? Doesn't this make it transparent?

Also we can perhaps log a warning if the deprecated option is used: https://github.com/open-feature/flagd/pull/1012/files#r1394782335

File: flagd-proxy/cmd/start.go (hunk 3 above)
Comment (craigpastro) [reply flagd-proxy/cmd/start.go:83]: Good idea. Deprecation warning log added. Thanks!

Comment (Kavindu-Dodan) [reply flagd-proxy/cmd/start.go:83]: @craigpastro Minor change - we can avoid the if clause of the warning log by moving it to this else clause.

**Comment by Kavindu-Dodan** [comment]
> Is this actually breaking with the change [here](https://github.com/open-feature/flagd/pull/1012/files#diff-b91a595f5d0f13cb46c493699f061cc8242e7cdef20cb73de82a24fe9b7b3b19R72-R79)? Doesn't this make it transparent?
> 
> Also we can perhaps log a warning if the deprecated option is used: https://github.com/open-feature/flagd/pull/1012/files#r1394782335

Yeah this is not a breaking change. We can modify the PR title if we all agree :)

**Review by thisthat — APPROVED** [review]

**Comment by toddbaert** [comment]
> > Is this actually breaking with the change [here](https://github.com/open-feature/flagd/pull/1012/files#diff-b91a595f5d0f13cb46c493699f061cc8242e7cdef20cb73de82a24fe9b7b3b19R72-R79)? Doesn't this make it transparent?
> > Also we can perhaps log a warning if the deprecated option is used: https://github.com/open-feature/flagd/pull/1012/files#r1394782335
> 
> Yeah this is not a breaking change. We can modify the PR title if we all agree :)

@craigpastro is there actually a breaking change here?

**Pull request merged**